tests/programs/custom/debug_test_trigger/debug_test_trigger.c

```diff
@@ -208,6 +211,58 @@ volatile uint8_t  g_some_data_bytes[4]     = {0xC0, 0xFF, 0xEB, 0xEE};
 volatile uint16_t g_some_data_halfwords[2] = {0xDEAD, 0xBEEF};
 volatile uint32_t g_some_data_word         = 0xC0DECAFE;
 
+/*
+ * has_umode_configured
+ *
+ * Check if we are running a core supporting umode
+ *
+ */
+uint32_t has_umode_configured(void) {
+  volatile uint32_t misa = 0x0;
+
+  __asm__ volatile (R"(
+    csrrs %[misa], misa, zero
+  )":[misa] "=r"(misa));
+
+  // CV32E40X does not support u-mode, skip test
+  switch ((misa & 0x00100000) > 0) {
+    case (0):
+      return 0;
+      break;
+    case (1):
+      ;; // Do nothing and continue execution
+      break;
+  }
+
+  return (1);
+}
+
+/*
+ * has_atomics_configured
+ *
+ * Check if we are running a core supporting A-extension
+ *
+ */
+uint32_t has_atomics_configured(void) {
+  volatile uint32_t misa = 0x0;
+
+  __asm__ volatile (R"(
+    csrrs %[misa], misa, zero
+  )":[misa] "=r"(misa));
+
+  // Conditionally support A-ext based on misa.A
+  switch (misa & 0x1) {
```

Comment:
`return (misa & 0x1);`?

Reply:
misa.A is bit 1

Reply:
This:
```
  switch (mycondition) {
    case (0):
      return 0;
      break;
    case (1):
      ;; // Do nothing and continue execution
      break;
  }

  return (1);
```

Should be the same as this?
```
  return mycondition;
```

Reply:
I guess I've been overcomplicating things here - well spotted